modules/datadict/php/datadict.class.inc

```diff
@@ -113,16 +113,149 @@ class Datadict extends \NDB_Menu_Filter
+            || ($_SERVER['REQUEST_METHOD'] === 'POST' && !isset($_POST['filter']))
+        ) {
+            // If it's a GET request (ie just reloading the page with
+            // previous filters or clicking on a sort column, use
+            // the filters that were saved into the session.
+            // OR a post request with no new defined filters (ie a second
+            // form on the page)
+            $savedFilters = $_SESSION['State']->getProperty('filter');
+            $this->_setFilters($savedFilters);
+
+            // Also set the form defaults based on the saved query, so that
+            // the filters values remain populated correctly in the front-end,
+            // not just in the query..
+            $defaults = $this->_getDefaults();
+            if (is_array($savedFilters)) {
+                $this->_setDefaults(array_merge($defaults, $savedFilters));
+            } else {
+                $this->_setDefaults($defaults);
+            }
+        } else if ($_SERVER['REQUEST_METHOD'] === 'POST'
+            && isset($_POST['filter'])
+        ) {
+            // If it's a POST, set the filters based on the
+            // request and save them into the session so that future
+            // GET requests have access to them.
+            $newFilters = $this->_setFilters($_POST);
+            if (!empty($newFilters)) {
+                $_SESSION['State']->setProperty('filter', $newFilters);
+            }
+        }
+
+        // 3. Set the search keyword (if applicable)
+        $key = null;
+        if (isset($_REQUEST['keyword'])) {
+            // It was included in the request, so save it.
+            $key = $_REQUEST['keyword'];
+            // Check if any quoted keyword
+            if (preg_match_all(
+                '\"[\w\s]+\"|\\\'[\w\s]+\\\'|[\w\s]+',
+                $key,
+                $keywords
+            )
+            ) {
+                $keywords = array_filter(
+                    array_map(
+                        function ($item) {
+                            return trim($item, ' ');
+                        },
+                        $keywords[0]
+                    )
+                );
+            }
+            if (is_array($keywords)) {
+                // Quoted keyword(s) exist.
+                $keywords_extra = array();
+                foreach ($keywords as &$keyword) {
+                    if (preg_match('\"[\w\s]+\"', $keyword)) {
+                        // Handle the quoted keyword.
+                        $keyword = '(?i)^' . $keyword . '$';
+                    } else if (preg_match('/\\s/', $keyword)) {
+                        // Handle multiple keywords.
+                        $split_keywords = explode(' ', $keyword);
+                        foreach ($split_keywords as &$split_keyword) {
+                            $split_keyword = '.*'.$split_keyword;
+                        }
+                        unset($split_keyword);
```

Comment:
Not sure this is necessary since you are not using this variable after this anyway

Reply:
Fair enough.